pwa/src/components/PatientInfoForm.tsx

```diff
@@ -0,0 +1,282 @@
+import React, { ReactElement, useState } from "react";
+import {
+  InputItem,
+  List,
+  TextareaItem,
+  Button,
+  Radio,
+  WhiteSpace,
+} from "antd-mobile";
+import "../scss/PatientInfoForm.scss";
+
+const RadioItem = Radio.RadioItem;
+
+export default function Form(): ReactElement {
+  // State Hooks for Radio Buttons
+  // useState returns a value, and a function to set that value
+  const [sex, setSex] = useState(null as Number); // Default: null
+  const [albendazole, setAlbendazole] = useState(0); // Default: No
+  const [firstVisit, setFirstVisit] = useState(0); // Default: No
+  const [pregnant, setPregnant] = useState(0); // Default: No
+  const [planning, setPlanning] = useState(0); // Default: No
+  const [bloodPressureValue, setBloodPressureValue] = useState("");
+
+  // Data for Radio Buttons
+  const sexValues = [
+    { value: 0, label: "Male" },
+    { value: 1, label: "Female" },
+  ];
+
+  const radioYesNo = [
+    { value: 0, label: "No" },
+    { value: 1, label: "Yes" },
+  ];
+
+  /* istanbul ignore next */
+  const onSubmit = () => {
+    if (process.env.NODE_ENV === "development") {
+      // This shows how to access the values of the form
+      const firstNameField: HTMLInputElement = document.getElementById(
+        "firstname"
```

Comment:
Is there an easy way to avoid hardcoding this "firstname" id in multiple locations @meghanmc or @bmitchinson ?
Other location: https://github.com/Haiti-Distributed-Healthcare-System/hdhs/blob/8cd1e4462476fb4da7184f9a4930795700e15d50/pwa/src/components/PatientInfoForm.tsx#L90

Reply:
Could we potentially use a central state object like this:
https://reactjs.org/docs/forms.html
```javascript
class NameForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {value: ''};

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({value: event.target.value});
  }

  handleSubmit(event) {
    alert('A name was submitted: ' + this.state.value);
    event.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>
          Name:
          <input type="text" value={this.state.value} onChange={this.handleChange} />
        </label>
        <input type="submit" value="Submit" />
      </form>
    );
  }
}
```
or would we have to shift the entire component to be a class component instead of a functional component?

Reply:
There seems to be a couple pieces of misunderstanding between us here, let's get on the same page, thanks for bringing this up 👍

Regarding the "hardcoding", we assign an id called `firstname` so that we can avoid "hardcoding".  Hardcoding imo for this would be something like "grab the value from the 4rd form element". To avoid that, we give it an id, so that we can reference it consistently, no matter what else is added to the form in any order. This method isn't too brittle and is resistant to change.

There is very little a class component can do that a functional component can not. Either a class or functional component could access a "central state object". Regardless of what we would want to do with state, a conversion between the two component formats would not be needed.

The term "central state" is confusing to me, as we're currently implementing what I believe to be "central state" in another in-progress ticket. In my experience the term "central state" is used to describe state that exists outside the component tree, but can be used by every component, which is how we're using [easy-peasy as described in this in-progress ticket](https://github.com/Haiti-Distributed-Healthcare-System/hdhs/issues/55).

The link you posted is an example of storing the input value in what I call the "local component state", as it doesn't exist outside the component.

After the above ticket has been completed, paitentinfo will store it's values in central state, and eventually after all forms are built, when we do our final submission of the entered date, we'll submit everything collected in central state, and then clear it for the next form entry. 

Along the way, each form component might use a bit of local state to enable easier validation. This way things are cleaned / errors are shown before we insert into central state. Patient info currently uses local state a bit to validate things like bloodpressue, that's how I addressed your review suggestion about it. [Example](https://github.com/Haiti-Distributed-Healthcare-System/hdhs/pull/66/commits/465c99790062ffef7f669e472a06efcdf65969ca#diff-f9ef59cdaa4eae3877a5abacd000f145R22). 

If you see an example of `useState` , that's the equivalent of the class example you posted above.

Does this all checkout @meghanmc ?

Reply:
@bmitchinson , everything you said matches my understanding. 

@abpwrs , if it helps - the entire onSubmit function will be replaced by a cleaner version that accesses each form field and sets it in what Ben is referring to as "central state". This entire function is included in the PR simply so we have an example of accessing the data fields once we add the central state. From how I am understanding your comment, I think it will not be a concern once we have added the other functionality.

Reply:
Sounds good to me!